sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -569,12 +586,17 @@ def _create_extra_packages(extra_packages, temp_dir):
 
       if not os.path.isfile(package):
         if Stager._is_remote_path(package):
-          # Download remote package.
-          _LOGGER.info(
-              'Downloading extra package: %s locally before staging', package)
-          _, last_component = FileSystems.split(package)
-          local_file_path = FileSystems.join(staging_temp_dir, last_component)
-          Stager._download_file(package, local_file_path)
+          if Stager._should_download_remote_extra_packages(package):
+            # Download remote package.
+            _LOGGER.info(
+                'Downloading extra package: %s locally before staging', package)
+            _, last_component = FileSystems.split(package)
+            local_file_path = FileSystems.join(staging_temp_dir, last_component)
+            Stager._download_file(package, local_file_path)
+          else:
+            remote_packages.append(package)
+            _LOGGER.info(
+                'Deferring download of extra package: %s to workers', package)
```

Comment:
This is potentially a breaking change for some folks (if their retrieval service doesn't have the right networking/permissions but their stager does) so it's worth documenting in the release notes (https://github.com/apache/beam/blob/master/CHANGES.md).

Reply:
Will do, thanks for the reminder. Is there anywhere else I should document this?

Reply:
The text of CHANGES.md eventually makes its way into the release notes, so that should be sufficient.